solver/src/solver/http_solver/model.rs

```diff
@@ -57,20 +58,20 @@ pub struct TokenInfoModel {
 #[derive(Debug, Serialize)]
 pub struct CostModel {
     pub amount: u128,
-    pub token: String,
+    pub token: H160,
 }
 
 #[derive(Debug, Serialize)]
 pub struct FeeModel {
     pub amount: u128,
-    pub token: String,
+    pub token: H160,
 }
 
 #[derive(Debug, Deserialize)]
 pub struct SettledBatchAuctionModel {
     pub orders: HashMap<String, ExecutedOrderModel>,
     pub uniswaps: HashMap<String, UpdatedUniswapModel>,
```

Comment:
Order IDs and AMM IDs should probably not be strings either right?

Reply:
They are just indices. I would prefer to leave this as it is not terribly harmful and these "indices" are essentially meaningless to us. The solver team wants these indices as strings. If you really want to change it, I would propose using usize since if you look at how these maps are made it is really just enumerate and use the index of the enumeration as the key here.

Reply:
> They are just indices

My issue is that the type `String` gives absolutely no indication as to what is being used for the key. For example, for the orders, a UID makes perfect sense as a key, and for pools, the pool address also makes perfect sense for a key. The issue is that there is an implicit format for the key that could be better expressed by choosing a more appropriate type. For example:
```rust
let mut map = HashMap::new();
map.insert(42usize, true);
println!("{}", serde_json::to_string(&map).unwrap()) // {"42":true}
```

This means that using `HashMap<usize, _>` whould output the exact same JSON while keeping a more concrete key type.

Other options that we can explore are using the short order UID (like in the explorer) as well as the AMM address which might make the instances slightly more debug-able.

Reply:
I think we are going to have a hard time getting the solver team to agree to all of this, but it is certainly worth a try. @marcovc, @twalth3r how would you feel about not using indices as keys for the orders and amms? I think the short orderUid is not a great idea because of potential collision. If the complete orderId is too much I would prefer to leave it as integer string indices and change our code internally from `String` to `usize`

Reply:
Ok, for now I have changed all the outgoing dictionary keys from String to usize. We are still accepting incoming settlement information as String and parsing back to usize where appropriate for comparison. I hope this can be the happy medium we seek for having good structures on one side, but not too bulk on the other.

Reply:
AFAIK `HashMap<usize, _>` keys are always strings, and shouldn't need manual parsing.

https://github.com/akubera/bigdecimal-rs/blob/fc797c302f4947e429cc331febc66fe3c7badc5e/Cargo.toml#L16

Reply:
Note that there is a `serde` feature on the crate to add serialization support:

```rust
let parsed = serde_json::from_str::<HashMap<usize, bool>>(r#"
    {
        "42": true
    }
"#);
println!("{:?}", parsed);
```

Reply:
Thats fine with me. In that case I will change all the incoming information to also be usize. Just wasn't sure about that.

Reply:
> I think we are going to have a hard time getting the solver team to agree to all of this

My point being that using `usize` here instead of `String` has no effect on the output JSON but gives us stricter typing on the Rust side. In other words, it would not require a change from on the HTTP solver side.